client/components/TestIframe/index.jsx

```diff
@@ -98,6 +103,11 @@ class TestIframe extends Component {
         // inside the iframe
         window.addEventListener('message', this.handleResultsMessage);
 
+        // Before hydrating, save local copy of serialized state with no results
+        // This is a short cut to tell whether the form has been edited
+        await this.waitForTestHarnessReload();
+        this.emptyForm = JSON.stringify(this.serializeForm());
```

Comment:
This is not always the "emptyForm" of the test you currently see in the iframe. The problem is we re-render when we have a new "testFile" coming in from the props, but maybe componentDidMount is not called again? But in this case, the tests are not be rehydrated. I'm not sure how I broke this in the refactor but I've run out of time to look into it now :(